pkg/controller/master/setting/handler.go

```diff
@@ -89,16 +89,15 @@ func (h *Handler) settingOnChanged(_ string, setting *harvesterv1.Setting) (*har
 			}
 			return setting, err
 		}
-
-		toUpdate := setting.DeepCopy()
-		if toUpdate.Annotations == nil {
-			toUpdate.Annotations = make(map[string]string)
-		}
-		toUpdate.Annotations[util.AnnotationHash] = currentHash
-		return setting, h.setConfiguredCondition(toUpdate, nil)
+		break
```

Comment:
using `break` to reconcile the setting doesn't look correct to me, prefer to add the resync method to those required settings accordingly.

Reply:
I think what we do here is not reconcile the setting. One setting can only be mapped to one `syncerFunc`. If we don't have error from `syncer`, we can just leave the for loop and update the setting's annotation and condition. If there is an error, we write it in condition and return.

https://github.com/harvester/harvester/blob/24240d896c1add07de72bfb4fb3d7bd446a6dba0/pkg/controller/master/setting/handler.go#L86-L91